calculator/main.py

```diff
@@ -25,9 +26,21 @@ def perform_operation(calc, operation, params):
     :param operation: String with the operation name. ie: 'add'
     :param params: Tuple containing the list of nums to operate with.
                    ie: (1, 2, 3, 4.5, -2)
+                   
     """
-    pass
-
+    if operation not in calc['operations'].keys():
+        raise InvalidOperation("Given operation is invalid.")
+
+ 
+    for element in params:
+        if type(element) == str:
```

Comment:
You're checking 1 parameter that can't be used... But I could send you for example a datetime. Isn't it easier to check for the parameters that you CAN use? Do a whitelist check instead of a blacklist check? Again, remember me on Saturday and we can talk about it.